templates/tftmpl/golden_test.go

```diff
@@ -458,13 +477,83 @@ func TestNewFiles(t *testing.T) {
 			input := tc.Input
 			input.init()
 			b := new(bytes.Buffer)
+
 			err := tc.Func(b, tc.Name, &input)
 			require.NoError(t, err)
 			checkGoldenFile(t, tc.Golden, b.String())
 		})
 	}
 }
 
+// TestVariablesTFVarsDetailed performs a more detailed tests on what is written to the variables.auto.tfvars file.
+// This is required since the order of fields cannot be guaranteed, and therefore a simple file comparison will
+// not suffice.
+func TestVariablesTFVarsDetailed(t *testing.T) {
+	task := Task{
+		Description: "user description for task named 'test'",
+		Name:        "test",
+		Source:      "namespace/consul-terraform-sync/consul//modules/test",
+		Version:     "0.0.0",
+	}
+
+	testCases := []struct {
```

Comment:
Does this have to be in a table?

Reply:
I ended up removing this test now that I sort the keys I can do the check in the TestNewFiles